src/iface/ethernet.rs

```diff
@@ -163,24 +166,25 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
     fn socket_ingress(&mut self, sockets: &mut SocketSet, timestamp: u64) -> Result<bool> {
         let mut processed_any = false;
         loop {
-            let frame =
-                match self.device.receive(timestamp) {
-                    Ok(frame) => frame,
-                    Err(Error::Exhausted) => break, // nothing to receive
-                    Err(err) => return Err(err)
-                };
-
-            let response =
-                match self.process_ethernet(sockets, timestamp, &frame) {
-                    Ok(response) => response,
-                    Err(err) => {
-                        net_debug!("cannot process ingress packet: {}", err);
-                        return Err(err)
-                    }
-                };
-            processed_any = true;
+            let &mut Self { ref mut device, ref mut inner } = self;
```

Comment:
I'm curious, is this really the only way to write this? Or would direct access to `self.device` and `self.inner` also work? No objection to this pattern anyway.

Reply:
I don't know of a better way. Direct access to `self.*` does not work, since the compiler then tries to borrow the complete `self` in the closure.